Finished Badge component with new examples

ericvalcik opened the pull request:
This PR should resolve [this issue](https://github.com/webscopeio/mailingui-web/issues/89).

Please let me know if you have any comments @jvorcak @iamhectorsosa.

iamhectorsosa reviewed (commented):
Looking nice!

Looking at those dots*, flex has limited support on email clients. How did it go with the tests?

Also any chance we can see if we can use a table to arrange these examples a bit better? The smaller examples seem to be aligned to the right. Maybe they all are, but the bigger ones look a bit more centered in the iframe.

ericvalcik commented:
I addressed the issues that arose and I also changed the `<ComponentExample>` component a little bit, so we can more clearly show the badges' examples.

@iamhectorsosa check it out and tell me what you think.

iamhectorsosa commented on src/app/(with-navbar)/components/[type]/page.tsx:
```diff
@@ -55,7 +55,11 @@ export default async function ComponentPage({
       <div className="mt-8 space-y-16 md:mt-16">
         {component.examples.map(({ ...example }, index) => (
-          <ComponentExample key={index} {...example} />
+          <ComponentExample
+            key={index}
+            {...example}
+            height={type === "badges" ? 120 : 350}
+          />
```
_Food for thought_: Ideal would be to have this dynamically set to the iframe's content height, much like they do in TailwindUI: [Newsletter Sections](https://tailwindui.com/components/marketing/sections/newsletter-sections). Notice how each of the components have different heights.

This is not a bad solution though. Although, we would probably have to create mappings for all of the components. 

Let's move forward for now and if you need to do this again in future PRs, let's extract this logic to make it cleaner and easier to maintain.

ericvalcik replied:
Good idea, I will work on this and include it in the next PR (the one where I implement Buttons).

iamhectorsosa commented on src/examples/components/badges/LargeBadges.tsx:
```diff
@@ -1,44 +1,47 @@
+          <span style={{ padding: 10 }} />
           <Badge variant="danger" size="lg">
             Badge
           </Badge>
-        </span>
-        <span style={{ padding: 10 }}>
+          <span style={{ padding: 10 }} />
           <Badge variant="success" size="lg">
             Badge
           </Badge>
-        </span>
-        <span style={{ padding: 10 }}>
+          <span style={{ padding: 10 }} />
           <Badge variant="info" size="lg">
             Badge
           </Badge>
-        </span>
-        <span style={{ padding: 10 }}>
+          <span style={{ padding: 10 }} />
           <Badge variant="warning" size="lg">
             Badge
           </Badge>
-        </span>
+        </Section>
```
Not against this at all. It works, but I don't if we could be a bit more consistent with this by using a [Column Primitive](https://react.email/docs/components/column) from React.email

I want us to think about this because this is very likely a pattern we will have to repeat in all or most examples and now it contains non-functional markup (the spans just for padding).

iamhectorsosa replied:
Thanks for trying this out

iamhectorsosa reviewed (approved):
Looking nice!

jvorcak reviewed (approved):
GJ!
